libraries/AP_Logger/AP_Logger.h

```diff
@@ -98,6 +98,77 @@ enum Log_Event : uint8_t {
     DATA_NOT_BOTTOMED = 166,
 };
 
+enum class LogErrorSubsystem : uint8_t {
```

Comment:
Would it be possible to adjust this so that these enums are defined by the vehicle?

Reply:
The point is to have these common across all vehicles - so we can use them in libraries, for example.

Reply:
https://github.com/ArduPilot/ardupilot/pull/10900/files#diff-deee08a6b031797222a9a8d2ef9bf407R538 <- that's an example of moving one of these into a library - the reason I had to make this PR in the first place :-)

Reply:
It just doesn't appear well thought out for common use, some of these are redundancy/overlap between those definitions that are common, and there remain a lot of definitions that are vehicle specific. Given that, I am left wondering what the actual point/benefit is.

Is there a phase 2 to this, or this is how we want these enumerations to look for some amount of time?

Reply:
The point is that that we can't make code common unless the libraries can create these log entry methods.

As stated earlier, I do have some interest in adding type-safety for the subsystem/code tuples we're passing in here.  It was too much to add to this PR, but the basic structure would be to have multiple `Write_Error` calls, one for each subsystem and having an enumeration of valid codes corresponding to the subsystem.

The immediate next step is to move the compass initialisation code up - which I can't do without this in (or we agree to lose the Write_Error calls, which is unlikely).

Reply:
As a side note, I think it will be easier for the GCS developers if we can have a common set of error messages instead of overlapping vehicle specific messages.  I think only the MP does the translation of error codes to human readable messages and it very likely shows the wrong messages for sub at the moment.

Reply:
Created an issue to track the lack of metadata:
https://github.com/ArduPilot/ardupilot/issues/10902

Reply:
> the libraries can create these log entry methods.

That's probably for the best.

And the behavioral change is the enumeration numbers for the error messages for sub?